Binary Eye 1.13.0 crashes immediately after scanning any barcode

Greetings, 

Firstly, thanks for all the work you have put into this project.  Since "Barcode Scanner" is no longer being maintained, I found this project while searching for another free software barcode scanner.  Unfortunately, the application crashes immediately after scanning any barcode; displaying "Unfortunately Binary Eye has stopped."  I have tried scanning CODE128, CODE93, QR, and Data Matrix.  All cause the crash as soon as the code is recognized.  

I am using Binary Eye version 1.13.0

Steps to Reproduce:  
1.  Open Binary Eye
2.  Point camera at barcode
3.  Observe crash

I have tried the software on two different devices, and both reproduce this issue:  

Samsung Galaxy S4 Mini (SCH-I435)
Android 4.4.2

Samsung Galaxy S5 (SM-G900V)
Android 5.0

Here is the stacktrace output from adb logcat:  

```
V/Vibrator(22016): Called vibrate(long) API - PUID: 10194, PackageName: de.markusfisch.android.binaryeye
V/Vibrator(22016): vibrate - PUID: 10194, PackageName: de.markusfisch.android.binaryeye, ms: 100, mag: -1
V/VibratorService(  934): vibrate - package: de.markusfisch.android.binaryeye, ms: 100, token: android.os.BinderProxy@43cf33a0
D/SensorManager(22016): unregisterListener ::   
W/dalvikvm(22016): Exception Ljava/util/regex/PatternSyntaxException; thrown while initializing Lc/a/a/a/a/d/b;
D/AndroidRuntime(22016): Shutting down VM
W/dalvikvm(22016): threadid=1: thread exiting with uncaught exception (group=0x41deeda0)
E/AndroidRuntime(22016): FATAL EXCEPTION: main
E/AndroidRuntime(22016): Process: de.markusfisch.android.binaryeye, PID: 22016
E/AndroidRuntime(22016): java.lang.ExceptionInInitializerError
E/AndroidRuntime(22016): 	at c.a.a.a.a.d.a.a.a()
E/AndroidRuntime(22016): 	at c.a.a.a.a.a.a()
E/AndroidRuntime(22016): 	at c.a.a.a.a.c.a()
E/AndroidRuntime(22016): 	at c.a.a.a.d.e.b()
E/AndroidRuntime(22016): 	at c.a.a.a.d.e.a()
E/AndroidRuntime(22016): 	at a.a.c.b.p.b()
E/AndroidRuntime(22016): 	at a.a.c.b.z.a()
E/AndroidRuntime(22016): 	at a.a.c.b.z.g()
E/AndroidRuntime(22016): 	at a.a.c.b.z.a()
E/AndroidRuntime(22016): 	at a.a.c.b.h.b()
E/AndroidRuntime(22016): 	at a.a.c.b.z.a()
E/AndroidRuntime(22016): 	at a.a.c.b.z.b()
E/AndroidRuntime(22016): 	at a.a.c.b.z.c()
E/AndroidRuntime(22016): 	at a.a.c.b.z.p()
E/AndroidRuntime(22016): 	at a.a.c.b.t.l()
E/AndroidRuntime(22016): 	at a.a.c.b.r.onStart()
E/AndroidRuntime(22016): 	at android.support.v7.app.m.onStart()
E/AndroidRuntime(22016): 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1177)
E/AndroidRuntime(22016): 	at android.app.Activity.performStart(Activity.java:5461)
E/AndroidRuntime(22016): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2386)
E/AndroidRuntime(22016): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2471)
E/AndroidRuntime(22016): 	at android.app.ActivityThread.access$900(ActivityThread.java:175)
E/AndroidRuntime(22016): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1308)
E/AndroidRuntime(22016): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/AndroidRuntime(22016): 	at android.os.Looper.loop(Looper.java:146)
E/AndroidRuntime(22016): 	at android.app.ActivityThread.main(ActivityThread.java:5602)
E/AndroidRuntime(22016): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(22016): 	at java.lang.reflect.Method.invoke(Method.java:515)
E/AndroidRuntime(22016): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1283)
E/AndroidRuntime(22016): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1099)
E/AndroidRuntime(22016): 	at dalvik.system.NativeStart.main(Native Method)
E/AndroidRuntime(22016): Caused by: java.util.regex.PatternSyntaxException: Unrecognized backslash escape sequence in pattern near index 26:
E/AndroidRuntime(22016): ^BEGIN:(VCARD|VEVENT)(?:\R.+?:[\s\S]+?)+?\REND:\1\R?$
E/AndroidRuntime(22016):                           ^
E/AndroidRuntime(22016): 	at java.util.regex.Pattern.compileImpl(Native Method)
E/AndroidRuntime(22016): 	at java.util.regex.Pattern.compile(Pattern.java:411)
E/AndroidRuntime(22016): 	at java.util.regex.Pattern.<init>(Pattern.java:394)
E/AndroidRuntime(22016): 	at java.util.regex.Pattern.compile(Pattern.java:374)
E/AndroidRuntime(22016): 	at d.i.o.<init>()
E/AndroidRuntime(22016): 	at c.a.a.a.a.d.b.<clinit>()
E/AndroidRuntime(22016): 	... 31 more
W/ActivityManager(  934):   Force finishing activity de.markusfisch.android.binaryeye/.activity.MainActivity
W/ActivityManager(  934):   Force finishing activity de.markusfisch.android.binaryeye/.activity.CameraActivity
D/CrashAnrDetector(  934): processName: de.markusfisch.android.binaryeye
D/CrashAnrDetector(  934): broadcastEvent : de.markusfisch.android.binaryeye data_app_crash
```

Thank you for your help.

--- a/app/src/main/kotlin/de/markusfisch/android/binaryeye/actions/vtype/VType.kt
+++ b/app/src/main/kotlin/de/markusfisch/android/binaryeye/actions/vtype/VType.kt
@@ -1,8 +1,9 @@
 package de.markusfisch.android.binaryeye.actions.vtype
 
 object VTypeParser {
+	private const val BACKSLASH_R_LEGACY = "(?:\u000D\u000A|[\u000A\u000B\u000C\u000D\u0085\u2028\u2029])"
 	private val vTypeRegex =
-		"""^BEGIN:(VCARD|VEVENT)(?:\R.+?:[\s\S]+?)+?\REND:\1\R?$""".toRegex(RegexOption.IGNORE_CASE)
+		"""^BEGIN:(VCARD|VEVENT)(?:$BACKSLASH_R_LEGACY.+?:[\s\S]+?)+?\REND:\1$BACKSLASH_R_LEGACY?$""".toRegex(RegexOption.IGNORE_CASE)
 	private val propertyRegex = """^(.+?):([\s\S]*?)$""".toRegex(RegexOption.MULTILINE)
 
 	fun parseVType(data: String): String? =

--- a/app/src/main/kotlin/de/markusfisch/android/binaryeye/actions/vtype/VType.kt
+++ b/app/src/main/kotlin/de/markusfisch/android/binaryeye/actions/vtype/VType.kt
@@ -3,7 +3,7 @@ package de.markusfisch.android.binaryeye.actions.vtype
 object VTypeParser {
 	private const val BACKSLASH_R_LEGACY = "(?:\u000D\u000A|[\u000A\u000B\u000C\u000D\u0085\u2028\u2029])"
 	private val vTypeRegex =
-		"""^BEGIN:(VCARD|VEVENT)(?:$BACKSLASH_R_LEGACY.+?:[\s\S]+?)+?\REND:\1$BACKSLASH_R_LEGACY?$""".toRegex(RegexOption.IGNORE_CASE)
+		"""^BEGIN:(VCARD|VEVENT)(?:$BACKSLASH_R_LEGACY.+?:[\s\S]+?)+?${BACKSLASH_R_LEGACY}END:\1$BACKSLASH_R_LEGACY?$""".toRegex(RegexOption.IGNORE_CASE)
 	private val propertyRegex = """^(.+?):([\s\S]*?)$""".toRegex(RegexOption.MULTILINE)
 
 	fun parseVType(data: String): String? =

